hardware/arduino/avr/cores/arduino/WString.cpp

```diff
@@ -108,14 +110,14 @@ String::String(unsigned long value, unsigned char base)
 String::String(float value, unsigned char decimalPlaces)
 {
 	init();
-	char buf[33];
+	char buf[FLT_MAX_10_EXP + 4 + decimalPlaces]; // +4, one for: 10, -, ., \0
```

Comment:
On AVR `DBL_MAX_10_EXP` is `38` and `Serial.println(String(-DBL_MAX));` prints out `-340282350000000000000000000000000000000.00`.

Which is `43` characters in length, then you need an extra byte for the `\0`, which brings us to `44`.

`44 - 38 - 2 = 4`. Maybe, I was thinking the extra character was for the `1's` digit ...